### PR TITLE
WDS-125 extend select options menu

### PR DESCRIPTION
**Pull request by Ctepan**
add option groups with headers and select all buttons
make select options menu width same as trigger's one

### CODE REVIEW

File: .changeset/four-readers-battle.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@soramitsu-ui/ui': patch
+---
+
+**fix**(`SSelect`): add option groups with headers and select all buttons
```
Comment (0x009922) [inline .changeset/four-readers-battle.md:5]: It seems that it is a significant feature, and definitely not a fix.

Why not `feature`? Why `patch` and not `minor`?

File: packages/ui/src/components/Popover/SPopover.ts [hunk 2]
```diff
@@ -129,6 +129,10 @@ export default /* @__PURE__ */ defineComponent({
       type: [Number, String],
       default: 0,
     },
+    sameWidth: {
+      type: Boolean,
+      default: false,
+    },
```
Comment (0x009922) [inline packages/ui/src/components/Popover/SPopover.ts:135]:
```suggestion
    sameWidth: Boolean,
```

File: packages/ui/src/components/Select/SSelectBase.vue [hunk 3]
```diff
@@ -107,6 +113,7 @@ provide(SELECT_API_KEY, api)
           name="s-select-dropdown-transition"
           eager
           :wrapper-attrs="{ 'class': 'z-10' }"
+          :inner-wrapper-attrs="{ 'class': sameWidthPopper && 'w-full' }"
```
Comment (0x009922) [inline packages/ui/src/components/Select/SSelectBase.vue:116]: I am not sure if setting `:class="false"` is correct from the Vue side.

This is better, imo:

```suggestion
          :inner-wrapper-attrs="{ 'class': { 'w-full': sameWidthPopper } }"
```

File: packages/ui/src/components/Select/use-model.ts [hunk 4]
```diff
@@ -59,6 +62,10 @@ export function useSelectModel<T = any>({
     return selectedSet.value.has(value)
   }
 
+  function isGroupSelected(optionGroup: SelectOptionGroup<T>) {
+    return optionGroup.items.every((x) => selectedSet.value.has(x.value))
```
Comment (0x009922) [inline packages/ui/src/components/Select/use-model.ts:66]: I think it's better to re-use exising fn:

```suggestion
    return optionGroup.items.every((x) => isValueSelected(x.value))
```

File: packages/ui/src/components/Select/use-model.ts [hunk 5]
```diff
@@ -81,7 +88,26 @@ export function useSelectModel<T = any>({
...
+  function toggleGroupSelection(optionGroup: SelectOptionGroup<T>): void {
+    if (isGroupSelected(optionGroup)) {
+      const optionGroupSet = new Set(optionGroup.items.map((x) => x.value))
+      model.value = modelAsArray.value.filter((x) => !optionGroupSet.has(x))
+
+      return
+    }
+
+    model.value = [...new Set([...modelAsArray.value, ...optionGroup.items.map((x) => x.value)])]
+  }
```
Comment (0x009922) [inline packages/ui/src/components/Select/use-model.ts:100]: Wouldn't using `else` be more readable then empty `return`?

```suggestion
  function toggleGroupSelection(optionGroup: SelectOptionGroup<T>): void {
    if (isGroupSelected(optionGroup)) {
      const optionGroupSet = new Set(optionGroup.items.map((x) => x.value))
      model.value = modelAsArray.value.filter((x) => !optionGroupSet.has(x))
    } else {
      model.value = [...new Set([...modelAsArray.value, ...optionGroup.items.map((x) => x.value)])]
    }
  }
```

File: .changeset/four-readers-battle.md (hunk 1 above)
Comment (Ctepan) [reply .changeset/four-readers-battle.md:5]: It was in 'documentation' but not implemented. I fixed it

**Comment by 0x009922** [comment]
Still unresolved conversations there.

**Pull request merged**